src/modules/economy/eco.ts

```diff
@@ -1,21 +1,237 @@
+import { Loot } from "../../schemas/loot";
+import { getNationStartingBonus } from "../../helpers";
+import { Building } from "../../schemas/buildings";
+
 const collection = db.collection("Users")
 
 export async function getUser(interaction: Interaction): Promise<User> {
     let player = await collection.findOne<User>(
         {
-            userData: {
-                userId: interaction.user.id,
-            }
+            "userInfo.userId": String(interaction.user.id)
+        }
+    )
+    if(player == null || player == undefined || !player) {
+        return null
+    }
+    return player
+}
+export async function getUserByUsername(username: string): Promise<User> {
+    let player = await collection.findOne<User>(
+        {
+            "userCreds.userName": String(username).toLowerCase(),
         }
     )
     if(player == null || player == undefined || !player) {
         return null
     }
     return player
-}
\ No newline at end of file
+}
+export async function checkUserByUsername(username: string) {
+    const player = await getUserByUsername(username)
+    if (player == null) {
+        console.log(`NOT FOUND : \nUsername : ${username}`)
+        return false
+    } else {
+        console.log(`FOUND : \nUsername : ${username}`)
+        return true
+    }
+}
+export async function checkUserById(interaction: Interaction) {
+    const player = await getUser(interaction)
+    if (player == null) {
+        return false
+    } else {
+        return true
+    }
+}
+export async function createUser(interaction: Interaction, username: string, nationName: string, nationRegion: number, key: string): Promise<User> {
+    const bonus = getNationStartingBonus(nationRegion)
+    await collection.insertOne(
+        {
+            userInfo: {
+                userName: interaction.user.username,
+                userId: interaction.user.id
+            },
+            userCreds: {
+                userName: username.toLowerCase(),
+                userPassword: key,
+            },
+            userAvatar: zeroUser.userAvatar,
+            userData: {
+                nation: {
+                    id: interaction.user.id,
+                    name: nationName,
+                    level: zeroUser.userData.nation.level,
+                    xp: zeroUser.userData.nation.xp,
+                    size: zeroUser.userData.nation.size,
+                    region: nationRegion,
+                    loot: bonus,
+                    production: zeroUser.userData.nation.production,
+                    buildings: zeroUser.userData.nation.buildings,
+                    lastCollect: Date.now(),
+                }
+            },
+            userPurchases: zeroUser.userPurchases,
+            userCooldowns: zeroUser.userCooldowns,
+        },
+    )
+    const playerToReturn = await getUser(interaction)
+    return playerToReturn
+}
+export async function editNation(interaction: Interaction, nation: Nation): Promise<User> {
```

Comment:
You do not have to rebuild the entire nation object to edit a nation. Instead use the TS type `Partial`.
Then you can use $update to update the object partially.
```suggestion
//The nation param will now be recognised even with only a few fields present
export async function editNation(interaction: Interaction, nation: Partial<Nation>): Promise<User> {
```